src/providers/azure/mod.rs

```diff
@@ -216,7 +217,15 @@ impl Azure {
         // make sure the metadata service is compatible with our version
         azure
             .is_fabric_compatible(MS_VERSION)
-            .chain_err(|| "failed version compatibility check")?;
+            .map_err(|_| {
+                let user = Uid::current();
+                let root_or_not = Uid::is_root(user);
+                if !root_or_not {
+                    warn!("unable to reach Azure endpoints, please check whether firewall rules are blocking access to them");
+                }
+
+                "failed to get shared configuration: not found"
```

Comment:
I think it would be better to keep the closure parameter and propagate it here (beware of types).
That is, something like:
```
.map_err(|e| { ... ; e})
```

Reply:
Thank you so much for review. I tested `.map_err(|e| { ... ; e})` works and show the clear chain errors.